data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDB.java

```diff
@@ -68,6 +68,32 @@ public Environment updateEnvironment(Environment environment)
     return updateEnvironment(environment, null);
   }
 
+  @Override
+  public EnvironmentRevision getEnvironmentRevision(
+      @NonNull final EnvironmentId environmentId, @NonNull final String environmentRevisionId)
+      throws ResourceNotFoundException, InternalServiceException {
+    final String accountIdClusterEnvironmentName =
+        environmentId.generateAccountIdClusterEnvironmentName();
+    try {
+      final EnvironmentRevisionDDBRecord environmentRevisionRecord =
+          dynamoDBMapper.load(
+              EnvironmentRevisionDDBRecord.class,
+              accountIdClusterEnvironmentName,
+              environmentRevisionId);
+      if (environmentRevisionRecord == null) {
+        throw new ResourceNotFoundException(
+            ResourceType.ENVIRONMENT_REVISION, environmentRevisionId);
+      }
+      return environmentMapper.toEnvironmentRevision(environmentRevisionRecord);
+    } catch (final AmazonServiceException e) {
+      throw new InternalServiceException(
+          String.format(
+              "Could not load record with environment revision key %s and environment revision id %s",
```

Comment:
Can we log the environment name here as well? It may give the user more information

Reply:
The key here actually is an accountIDClusterEnvionrment hash key,  so it contains all the information including environment name.